File: app/drivers/chrome/src/persist.js
function if_per_set(ns, key, value) {
  var _ns = store.namespace(ns);
  _ns.set(key, value)
}

function if_per_del(ns, key) {
  var _ns = store.namespace(ns);
  _ns.remove(key);
}

function if_per_del_ns(ns) {
  var _ns = store.namespace(ns);
  _ns.clearAll();
}

function if_per_get(s, ns, key) {
  var _ns = store.namespace(ns);
  var res = _ns.get(key);
  if (res === undefined) {
    throw "This should never be reached. if_per_get returned undefined, it should return null";
  }

  int_dispatch([4, "int_per_get_res", s, ns, key, res]);
}

File: app/drivers/apple/flok-pod/Pod/Classes/Drivers/Persist.swift
@objc class FlokPersistModule : FlokModule {
    override var exports: [String] {
        return ["if_per_set:", "if_per_del:", "if_per_del_ns:", "if_per_get:"]
    }
    
    func if_per_set(args: [AnyObject]) {
        let ns = args[0] as! String
        let key = args[1] as! String
        let value = args[2] as! String
        
        NSUserDefaults.standardUserDefaults().setObject(value, forKey: nsk(ns, key))
        NSUserDefaults.standardUserDefaults().synchronize()
    }

    func if_per_del(args: [AnyObject]) {
        let ns = args[0] as! String
        let key = args[1] as! String
        NSUserDefaults.standardUserDefaults().removeObjectForKey(nsk(ns, key))
        NSUserDefaults.standardUserDefaults().synchronize()
    }

    func if_per_del_ns(args: [AnyObject]) {
        let ns = args[0] as! String
        
        let keys = NSUserDefaults.standardUserDefaults().dictionaryRepresentation().keys
        for e in keys {
            if e.rangeOfString(nsk(ns, nil)) != nil {
                NSUserDefaults.standardUserDefaults().removeObjectForKey(e)
            }
        }
        
        NSUserDefaults.standardUserDefaults().synchronize()
    }

    func if_per_get(args: [AnyObject]) {
        let s = args[0] as! String
        let ns = args[1] as! String
        let key = args[2] as! String
        
       let res =  NSUserDefaults.standardUserDefaults().objectForKey(nsk(ns, key)) ?? NSNull()
       engine.int_dispatch([4, "int_per_get_res", s, ns, key, res])
    }
}

//Creates a 'namespaced' key for a namespace+non-namespaced key pair
private func nsk(namespace: String, _ key: String?) -> String {
    return "\(namespace)://____\(key ?? "")"
}